lib/nexmo/namespace.rb

```diff
@@ -59,6 +61,28 @@ def request(path, params: nil, type: Get, &block)
       parse(response, &block)
     end
 
+    def persistent_request(to_array, path, params)
+      Net::HTTP.start(host) do |http|
+        @http = http
+        to_array.each do |to|
+          params[:params][:to] = to
+          # retry 5 times
```

Comment:
instead of having a 5 loop and break I'd suggest doing something like:
```ruby
attempts = 0
begin
  messages = request(path, params).messages
  case messages.first&.status
  when '0'
    logger.success("Message successfully sent to #{to}.")
  when '1'
    raise Nexmo::Error
  else
    @failed_deliveries << result.messages
  end
rescue Nexmo::Error => error
  if attempts <= 5
    attempts += 1
    sleep 1
    retry
  end
end
```